File: app/src/main/java/org/metabrainz/mobile/api/data/search/entity/Artist.java
package org.metabrainz.mobile.api.data.search.entity;

import com.google.gson.annotations.SerializedName;

public class Artist {

    @SerializedName("id")
    private String mbid;
    private String name;
    private String country;
    private String disambiguation;
    private String type;
    @SerializedName("sort-name")
    private String sortName;
    private Area area;
    @SerializedName("begin-area")
    private Area beginArea;
    @SerializedName("end-area")
    private Area endArea;

    public Artist(String mbid, String name, String country, String disambiguation) {
        this.mbid = mbid;
        this.name = name;
        this.country = country;
        this.disambiguation = disambiguation;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getSortName() {
        return sortName;
    }

    public void setSortName(String sortName) {
        this.sortName = sortName;
    }

    public Area getArea() {
        return area;
    }

    public void setArea(Area area) {
        this.area = area;
    }

    public Area getBeginArea() {
        return beginArea;
    }

    public void setBeginArea(Area beginArea) {
        this.beginArea = beginArea;
    }

    public Area getEndArea() {
        return endArea;
    }

    public void setEndArea(Area endArea) {
        this.endArea = endArea;
    }

    public String getMbid() {
        return mbid;
    }

    public void setMbid(String mbid) {
        this.mbid = mbid;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDisambiguation() {
        return disambiguation;
    }

    public void setDisambiguation(String disambiguation) {
        this.disambiguation = disambiguation;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    @Override
    public String toString() {
        return "Artist{" +
                "mbid='" + mbid + '\'' +
                ", name='" + name + '\'' +
                ", country='" + country + '\'' +
                ", disambiguation='" + disambiguation + '\'' +
                '}';
    }
}

File: app/src/main/java/org/metabrainz/android/model/mbentity/Artist.kt
package org.metabrainz.android.model.mbentity

import com.google.gson.annotations.SerializedName
import org.metabrainz.android.model.entities.LifeSpan
import org.metabrainz.android.model.entities.Link
import java.util.*

class Artist : org.metabrainz.android.model.mbentity.MBEntity() {
    var name: String? = null
    var country: String? = null
    var type: String? = null

    @SerializedName("sort-name")
    var sortName: String? = null
    var area: org.metabrainz.android.model.mbentity.Area? = null

    @SerializedName("begin-area")
    var beginArea: org.metabrainz.android.model.mbentity.Area? = null

    @SerializedName("end-area")
    var endArea: org.metabrainz.android.model.mbentity.Area? = null

    @SerializedName("life-span")
    var lifeSpan: LifeSpan? = null
    var gender: String? = null
    var relations: MutableList<Link> = ArrayList()
        set(relations) {
            this.relations.addAll(relations)
        }
    @JvmField
    val releases: MutableList<org.metabrainz.android.model.mbentity.Release> = ArrayList()
    fun getReleases(): List<org.metabrainz.android.model.mbentity.Release> {
        return releases
    }

    fun setReleases(releases: List<org.metabrainz.android.model.mbentity.Release>?) {
        this.releases.addAll(releases!!)
    }

    fun setRelease(release: org.metabrainz.android.model.mbentity.Release, position: Int) {
        releases[position] = release
    }

    override fun toString(): String {
        return "Artist{" +
                "mbid='" + mbid + '\'' +
                ", name='" + name + '\'' +
                ", country='" + country + '\'' +
                ", disambiguation='" + disambiguation + '\'' +
                '}'
    }
}